src/main/java/sg/ncl/MainController.java

```diff
@@ -1532,6 +1535,91 @@ public String rejectJoinRequest(@PathVariable Integer teamId, @PathVariable Inte
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "The image '" + imageName + "' either does not exist or not found in your teams' list of images");
+                        return "redirect:/teams";
+                    case DETERLAB_OPERATION_FAILED_EXCEPTION:
+                        log.warn("Error in deleting image '{}' from team '{}' : operation failed on DeterLab", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                    case ADAPTER_CONNECTION_EXCEPTION:
+                        log.warn("Error in deleting  image '{}' from team '{}' : adapter connection error", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                    case ADAPTER_INTERNAL_ERROR_EXCEPTION:
+                        log.warn("Error in deleting image '{}' from team '{}' : adapter internal error", imageName, teamId);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                    default:
+                        log.warn("Error in deleting image '{}' from team '{}' : {}", imageName, teamId,  exceptionState);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                }
+            } else {
+                String sioMessage = new JSONObject(responseBody).getString("msg");
+
+                switch (sioMessage) {
+                    case "image still in use":
+                        log.warn("Error in deleting image '{}' from team '{}' : {}", imageName, teamId, sioMessage);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_FAILURE, "Image " + "\'" + imageName + "\'" + " is still in use or busy!");
+                        return "redirect:/teams";
+                    // curl command is ok but there is problem with rm command
+                    case "delete image OK from web but error when executing rm command to delete physical image":
+                        log.warn("Error in deleting image '{}' from team '{}' : {}", imageName, teamId, sioMessage);
+                        redirectAttributes.addFlashAttribute(MESSAGE_DELETE_IMAGE_WARNING, "Image " + "\'" + imageName + "\'" + " is successfully deleted. " +
+                                                                                        "However, " + ERR_SERVER_OVERLOAD);
+                        return "redirect:/teams";
+                    case "delete image OK from web but there is unknown error when deleting physical image":
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This case's code block is the same as the block for the case on line 1599. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1871)